src/MergeContext.js

```diff
@@ -454,6 +454,202 @@ class BranchPosition
     }
 }
 
+// Forward iterator for fields in the 'name:value' format.
+class FieldsTokenizer
+{
+    // input string and an array of allowed attributes
+    constructor(str, allowedAttributes) {
+        this._lines = str.split('\n');
+        // supported attributes
+        this._attributes = allowedAttributes;
+    }
+
+    // Searches for a line matching the 'attrName:attrValue' pattern,
+    // where attrName is one of the allowed attributes.
+    // All parsed lines are removed from the input.
+    // Returns the matched {attrName, attrValue} pair (or null).
+    nextField() {
+        while (!this.atEnd()) {
+            const line = this._lines.shift();
+            const pos = line.search(':');
+            if (pos >= 0) {
+                const name = line.substring(0, pos);
+                if (this._attributes.includes(name))
+                    return {name: name, value: line.substring(pos+1)};
+            }
+        }
+        return null;
+    }
+
+    // whether the input is empty
+    atEnd() {
+        return this._lines.length === 0;
+    }
+
+    // returns the unparsed (yet) string
+    remaining() {
+        return this._lines.join('\n');
+    }
+}
+
+// Converts the raw PR message into the staged commit message.
```

Comment:
"PR message" sounds odd/undefined, and we are using many PR parts, not just the PR title/description.

```suggestion
// Computes future commit message from raw PR
```